packages/suite/src/views/wallet/trading/buy/TradingBuyConfirm.tsx

```diff
@@ -1,26 +1,38 @@
 import { FormProvider } from 'react-hook-form';
 
+import { SelectedAccountLoaded } from '@suite-common/wallet-types';
+
+import { useSelector } from 'src/hooks/suite';
 import { useTradingBuyForm } from 'src/hooks/wallet/trading/form/useTradingBuyForm';
 import { TradingFormContext } from 'src/hooks/wallet/trading/form/useTradingCommonForm';
 import { UseTradingProps } from 'src/types/trading/trading';
+import { getProvidersInfoProps } from 'src/utils/wallet/trading/tradingTypingUtils';
+import { getTradeProvider } from 'src/utils/wallet/trading/tradingUtils';
 import { TradingContainer } from 'src/views/wallet/trading/common/TradingContainer';
 import { TradingSelectedOffer } from 'src/views/wallet/trading/common/TradingSelectedOffer/TradingSelectedOffer';
 
-const TradingBuyConfirmComponent = ({ selectedAccount }: UseTradingProps) => {
+const TradingBuyConfirmLoaded = ({ selectedAccount }: UseTradingProps) => {
     const tradingBuyContextValues = useTradingBuyForm({
         selectedAccount,
         pageType: 'confirm',
     });
 
+    const provider = getTradeProvider({
+        trade: tradingBuyContextValues.selectedQuote,
+        providerInfo: getProvidersInfoProps(tradingBuyContextValues),
+    });
+
     return (
         <TradingFormContext.Provider value={tradingBuyContextValues}>
             <FormProvider {...tradingBuyContextValues.methods}>
-                <TradingSelectedOffer />
+                <TradingContainer SectionComponent={TradingSelectedOffer} provider={provider} />
             </FormProvider>
         </TradingFormContext.Provider>
     );
 };
 
-export const TradingBuyConfirm = () => (
-    <TradingContainer SectionComponent={TradingBuyConfirmComponent} />
-);
+export const TradingBuyConfirm = () => {
+    const selectedAccount = useSelector(state => state.wallet.selectedAccount);
+
+    return <TradingBuyConfirmLoaded selectedAccount={selectedAccount as SelectedAccountLoaded} />;
```

Comment:
Avoid using type assertions (`as SelectedAccountLoaded`) here.


```suggestion
   if (selectedAccount.status !== 'loaded') {
        return null;
    }
    
    return <TradingBuyConfirmLoaded selectedAccount={selectedAccount} />;
```

Reply:
done